doc: releases: add more release highlights for 4.0

kartben opened the pull request:
This adds a few more entries to hopefully complete the list of
4.0 highlights + other cleanups (spelling/typos and removing empty sections)

https://builds.zephyrproject.io/zephyr/pr/81434/docs/releases/release-notes-4.0.html

kartben committed:
doc: releases: add more release highlights for 4.0

This adds a few more entries to hopefully complete the list of
4.0 highlights.

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

kartben committed:
doc: releases: fix Devicetree spelling

Spell Devicetree with a capital D

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

kartben committed:
doc: releases: drop empty driver subsections

Drop sections with no updates

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

kartben committed:
doc: releases: drop empty lib/subsystems subsections

Drop sections with no updates

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

kartben committed:
doc: releases: fix typos in MCUboot section

Fix minor typos

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

kartben committed:
doc: releases: drop empty subsections from migration guide

Drop sections with no updates

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

kartben committed:
doc: releases: fix galaxycore,gc2145 compatible

fix typo in compatible name

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

kartben committed:
doc: releases: use correct compatible for zephyr.gpio-steppers

The current name of the compatible is zephyr,gpio-steppers

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

kartben commented on doc/releases/release-notes-4.0.rst:
```diff
@@ -1220,14 +1224,8 @@ Libraries / Subsystems
 
-* Task Watchdog
-
 * POSIX API
```
note that POSIX API was intentionally kept to not clash with https://github.com/zephyrproject-rtos/zephyr/pull/81442

mmahadevan108 commented:
I think the API changes section could be updated to move everything either to `Removed API` or `Deprecated API` sub-sections

kartben committed:
doc: releases: re-arrange API Changes section

some entries were not properly put under either
Removed or Deprecated section. This commit
fixes that.

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

kartben commented:
> I think the API changes section could be updated to move everything either to `Removed API` or `Deprecated API` sub-sections

+1 - done

mmahadevan108 commented:
Can you please rebase
